<!-- The following command can be modified to run from terminal -->
<!-- roslaunch zerog_utilities stp.launch x:=2.237 y:=-2.034 z:=1.522 qx:=-0.002 qy:=-0.706 qz:=0.004 qw:=0.708 frame_id:=world child_frame_id:=CW namespace:=world_CW -->


<launch>


        <!-- Define arguments -->
        <arg name="x" default="2.237" />
        <arg name="y" default="-2.034" />
        <arg name="z" default="1.522" />
        <arg name="qx" default="-0.002" />
        <arg name="qy" default="-0.706" />
        <arg name="qz" default="0.004" />
        <arg name="qw" default="0.708" />
        <arg name="frame_id" default="world" />
        <arg name="child_frame_id" default="CW" />
        <arg name="namespace" default="default_namespace" />
    
        <!-- Use the arguments in the node -->
        <group ns="$(arg namespace)">
            <node
                pkg="tf2_ros"
                type="static_transform_publisher"
                name="$(arg frame_id)_$(arg child_frame_id)"
                args="$(arg x) $(arg y) $(arg z) $(arg qx) $(arg qy) $(arg qz) $(arg qw) $(arg frame_id) $(arg child_frame_id)"
            />
        </group>
</launch>
    
